torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -361,23 +367,25 @@ def to_sharded_post_forward_dtensor(self, tensor: torch.Tensor) -> DTensor:
             self._global_stride,
         )
 
-    def alloc_all_gather_output(self) -> None:
-        unsafe_alloc_storage(self.all_gather_output)
+    def alloc_all_gather_outputs(self) -> None:
+        unsafe_alloc_storage(self.all_gather_outputs[0])
 
-    def free_all_gather_output(self) -> None:
-        unsafe_free_storage(self.all_gather_output)
+    def free_all_gather_outputs(self) -> None:
+        unsafe_free_storage(self.all_gather_outputs[0])
```

Comment:
I assume `all_gather_outputs[0]` is `NF4Tensor`? Is this calling `NF4Tensor..untyped_storage().resize_`? or it's calling on inner tensors ?

Reply:
this question is more for clarification since the PR stack is validated

Reply:
This part will be for the non-extensions code path, where `len(self.all_gather_outputs) == 1` always.

In the next PR, we change `free_all_gather_outputs` to `free_unsharded_param`:
https://github.com/pytorch/pytorch/blob/262cecc0208b4f88c2ad8aad7d83a0c3c9779283/torch/distributed/_composable/fsdp/_fsdp_param.py#L445-L449

For default non-extensions case, the all-gather output and the unsharded parameter share the same tensor data, so freeing all-gather output vs. unsharded parameter is the same thing.

For extensions case, the all-gather outputs can either alias or not alias the unsharded parameter. Regardless, we choose to free the all-gather outputs and unsharded parameter altogether in `free_unsharded_param` to avoid having to check for any aliasing at runtime (which may complicate compiling it).

Reply:
thanks. now i understand why ``len(self.all_gather_outputs) == 1`` always